src/main.cpp

```diff
@@ -97,7 +97,8 @@ int main(int argc, char** argv) {
     try {
       if (!limit_arg.empty()) {
         // Internally timeout is in milliseconds.
-        cl_args.timeout = 1000 * std::stof(limit_arg);
+        cl_args.timeout = std::chrono::milliseconds(
+          static_cast<int>(1000 * std::stof(limit_arg)));
```

Comment:
Thanks for the change. Something else occured to me: the cast here is limiting the max number of milliseconds one can use as a timeout to the max value of an `int`. This mean the max possible value is halved compared to the previous setup with an implicit cast to `unsigned`.

My understanding is that the actual limitation will depend on the number of bits used for `int` on a given platform. The standard says "at least 16" which is not much for a number of milliseconds. On the other hand, `std::chrono::milliseconds` is a helper type for `std::chrono::duration</*signed integer type of at least 45 bits*/, std::milli>`...

Reply:
Cool. Switched to `std::chrono::milliseconds::rep`.